### ISSUE TITLE
Create New Request Window Closes When Clicking Outside Without Warning

### ISSUE THREAD

**Issue by jeevankumarj25** [usebruno/bruno#2325]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

When creating a new request within a collection, the "Create Request" window closes immediately if the user clicks outside the window. This behavior occurs without any warning or confirmation, leading to potential loss of any data entered in the request form. This issue makes it challenging to copy and paste data from another browser window or document, disrupting the workflow.
![createNewRequest](https://github.com/usebruno/bruno/assets/118682799/cab080d0-2ef4-4422-9f4d-c464d5fe266d)

### Screenshots/Live demo link

1. Open the application and navigate to the section where collections are managed.
2. Create a new collection if one does not already exist.
3. Within the collection, initiate the creation of a new request.
4. In the "Create Request" window, enter a request name and HTTP method.
5. Click outside the "Create Request" window (anywhere on the screen outside the popup).
6. 
![createNewRequest](https://github.com/usebruno/bruno/assets/118682799/56fbe378-63d7-4706-bc00-e4326008cfe8)

**Comment by vayalildanie** [comment]
This is not a bug, I think this is a purposeful design choice?

**Comment by naman-bruno** [comment]
Yep, closing this since it's an intentional design choice. There are a few modals that don't currently follow this behavior, and we'll be updating them to ensure consistency across the app. Thanks!
